src/interfaces/swig/SGBase.i

```diff
@@ -373,12 +379,39 @@ namespace std {
 %include <shogun/base/Parallel.h>
 %include <shogun/lib/StoppableSGObject.h>
 
-#ifdef SWIGPYTHON
 namespace shogun
 {
-
     %extend CSGObject
     {
+        std::vector<std::string> parameter_names() const {
+            std::vector<std::string> result;
+            for (auto const& each: $self->get_params()) {
+                result.emplace_back(each.first);
```

Comment:
just curious, I always thought that emblace is more for cases where one calls a constructor when placing the element to avoid creating a temp variable. Is there a benefit here as well?

Reply:
Ah sorry, that's from me copying code to handle std::reference_wrapper, that should be push_back